flutter/lib/src/screenshot/recorder.dart

```diff
@@ -47,9 +61,17 @@ class ScreenshotRecorder {
       // On Android, the desired resolution (coming from the configuration)
       // is rounded to next multitude of 16 . Therefore, we scale the image.
       // On iOS, the screenshot resolution is not adjusted.
-      final srcWidth = renderObject.size.width;
-      final srcHeight = renderObject.size.height;
-      final pixelRatio = config.getPixelRatio(srcWidth, srcHeight);
+      final srcWidth = renderObject.size.width.toInt();
+      final srcHeight = renderObject.size.height.toInt();
+
+      // In Session Replay the target size is already set and should not be changed.
+      // For Screenshots, we need to calculate the target size based on the quality setting.
+      config.targetHeight ??=
+          config.quality.calculateHeight(srcWidth, srcHeight);
+      config.targetWidth ??= config.quality.calculateWidth(srcWidth, srcHeight);
+
+      var pixelRatio =
+          config.getPixelRatio(srcWidth.toDouble(), srcHeight.toDouble());
```

Comment:
I understand you've made these changes to fix the issue with screenshots not filling the whole space. I'm pretty confident these changes aren't necessary. The pixel ratio is actually what needs to be coming based on the config. And it must be a single value because we don't want to skew the image.

Reply:
I refactored it, based on your input. I think it is now way cleaner as before.